vizro-core/tests/unit/vizro/models/_action/test_action.py

```diff
@@ -113,18 +116,43 @@ def test_inputs_invalid(self, inputs, test_action_function):
         ],
     )
     def test_outputs_invalid(self, outputs, test_action_function):
-        with pytest.raises(ValidationError, match=".* value is not a valid list .*"):
+        with pytest.raises(ValidationError, match="value is not a valid list"):
             Action(function=test_action_function, inputs=[], outputs=outputs)
 
+    @pytest.mark.parametrize("file_format", [None, "csv", "xlsx"])
+    def test_export_data_file_format_valid(self, file_format):
+        action = Action(id="action_test", function=export_data(file_format=file_format))
+        assert action.id == "action_test"
+        assert action.inputs == []
+        assert action.outputs == []
+
+    def test_export_data_file_format_invalid(self):
+        with pytest.raises(
+            ValueError, match='Unknown "file_format": invalid_file_format.' ' Known file formats: "csv", "xlsx".'
+        ):
+            Action(function=export_data(file_format="invalid_file_format"))
+
+    def test_export_data_xlsx_without_required_libs_installed(self, monkeypatch):
+        # Mock only if the 'importlib.util.find_spec(argument)' argument is "openpyxl" or "xlsxwriter".
+        monkeypatch.setattr(
+            importlib.util,
+            "find_spec",
+            lambda arg: None if arg in ["openpyxl", "xlsxwriter"] else importlib.util.find_spec(arg),
+        )
```

Comment:
Let's try to make this test more "black box" by doing it in a way that doesn't directly rely on the implementation. So basically try to monkeypatch at a higher level, something like this:
```
monkeypatch.setitem(sys.modules, "openpyxl", None)
monkeypatch.setitem(sys.modules, "xlswriter", None)
```